cli/src/main.rs

```diff
@@ -606,10 +606,10 @@ fn send_direct_request(
 
 	debug!("setup sender and receiver");
 	let (sender, receiver) = channel();
-	match direct_api.watch(jsonrpc_call, sender) {
+	thread::spawn(move || match direct_api.watch(&jsonrpc_call, &sender) {
```

Comment:
Moved thread spawning out of the watch function for better visibility and variability of watch handling

Reply:
What does this allow you to do, which wasn't possible before?

Reply:
Two things:
- wait for the thread to finish (join), 
- react upon an Err value when connecting to the client. Before, an error log was printed, but no error returned:
```rust
thread::spawn(move || {
			info!("attempting to connect to RPC");
			match connect(url, |out| WsClient {
				out,
				request: request.clone(),
				result: sender.clone(),
				do_watch: true,
			}) {
				Ok(c) => {
					info!("connect was successful");
					c
				},
				Err(_) => {
					error!("Could not connect to direct invocation server");
				},
			}
		});
```

Reply:
Hmm, I don't really like it. We make hidden boilerplate code visible. This is only acceptable for me if you absolutely need to wait for the threads to join, but then you could just return the thread handle that `thread::spawn` returns.

If you want to return an error, you can simply change the signature in the of the thread spawn to:

```
thread::spawn(move || -> FancyResult<T> {
});
```

Reply:
Moved thread spawning back to the watch command.